WebKit::WebPageProxy::takeFocus() should take a WebCore::FocusDirection

ddkilzer opened the pull request:
#### dc7bc53309b5eda8f53fd8771933f2b69c48f135
<pre>
WebKit::WebPageProxy::takeFocus() should take a WebCore::FocusDirection
<a href="https://bugs.webkit.org/show_bug.cgi?id=249958">https://bugs.webkit.org/show_bug.cgi?id=249958</a>
&lt;rdar://65494875&gt;

Reviewed by Alex Christensen.

Update WebPageProxy::takeFocus() to use the enum class
WebCore::FocusDirection type instead of uint32_t.

* Source/WebCore/page/FocusDirection.h:
- Fix formatting nit.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
- Add WebCore::FocusDirection enum class for IPC.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
- Remove &apos;using namespace WebCore;&apos; statement as drive-by
  clean-up.
(WebKit::WebPageProxy::takeFocus):
- Switch to use WebCore::FocusDirection type.
- Remove unneeded static_cast&lt;&gt; operators.
* Source/WebKit/UIProcess/WebPageProxy.h:
(WebKit::WebPageProxy::takeFocus):
- Switch to use WebCore::FocusDirection type.
* Source/WebKit/UIProcess/WebPageProxy.messages.in: Ditto.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::takeFocus):
- Remove unneeded static_cast&lt;&gt; operator.

Canonical link: <a href="https://commits.webkit.org/258415@main">https://commits.webkit.org/258415@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3219b9d9b5f78ccf7597b37fec2813efccaa7ed5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101899 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11046 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34971 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111232 "Failed to checkout and rebase branch from PR 8107") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/171438 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105879 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12009 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1961 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/108987 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107680 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/92455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/94305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/23887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/4631 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/25369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4718 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/1809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/10796 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/44856 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6470 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3038 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/page/FocusDirection.h:
```diff
@@ -38,3 +40,20 @@ enum class FocusDirection: uint8_t {
+
+namespace WTF {
+
+template<> struct EnumTraits<WebCore::FocusDirection> {
```
This would be even better in WebCoreArgumentCoders.serialization.in instead.

ddkilzer replied:
Really curious how moving this away from where the enum is defined is better?

Or is there a mechanism to catch when they get out-of-sync?

Will give this a go.

Commit:
WebKit::WebPageProxy::takeFocus() should take a WebCore::FocusDirection
https://bugs.webkit.org/show_bug.cgi?id=249958
<rdar://65494875>

Reviewed by Alex Christensen.

Update WebPageProxy::takeFocus() to use the enum class
WebCore::FocusDirection type instead of uint32_t.

* Source/WebCore/page/FocusDirection.h:
- Fix formatting nit.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
- Add WebCore::FocusDirection enum class for IPC.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
- Remove 'using namespace WebCore;' statement as drive-by
  clean-up.
(WebKit::WebPageProxy::takeFocus):
- Switch to use WebCore::FocusDirection type.
- Remove unneeded static_cast<> operators.
* Source/WebKit/UIProcess/WebPageProxy.h:
(WebKit::WebPageProxy::takeFocus):
- Switch to use WebCore::FocusDirection type.
* Source/WebKit/UIProcess/WebPageProxy.messages.in: Ditto.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::takeFocus):
- Remove unneeded static_cast<> operator.

Canonical link: https://commits.webkit.org/258415@main
